bitcoin/src/psbt/mod.rs

```diff
@@ -133,6 +214,40 @@ impl Psbt {
         tx
     }
 
+    #[inline]
+    fn internal_extract_tx_with_checks(
+        self,
+        max_fee_rate: FeeRate,
+    ) -> Result<Transaction, ExtractTxError> {
+        let fee = match self.fee() {
+            Ok(fee) => fee,
+            Err(Error::MissingUtxo) =>
+                return Err(ExtractTxError::MissingInputValue {
+                    transaction: self.internal_extract_tx(),
+                }),
+            Err(Error::NegativeFee) => return Err(ExtractTxError::SendingTooMuch { psbt: self }),
+            Err(Error::FeeOverflow) =>
+                return Err(ExtractTxError::AbsurdFeeRate {
+                    fee_rate: FeeRate::MAX,
+                    transaction: self.internal_extract_tx(),
+                }),
+            _ => unreachable!(),
```

Comment:
This is technically correct but doesn't sit well with me, I'm ok to ack but wanted to flag that the correct solution is to first fix all the error handling in `psbt` then do this PR - but that is loads of work (to remove the general error type).

Reply:
I wish there was a better way to link the two types so that if we return a new variant from the fee function this would fail to compile...

Reply:
I'll resolve this for now.

Reply:
Seconded that this makes me uneasy.